### ISSUE TITLE
Call is not yet supported for this function: public constructor Array<T>

### ISSUE THREAD

**Issue by SteffenGorenflo** [cbeust/klaxon#110]
If I define a property in my data class with Array<T> like this:

```Kotlin
data class FooBar(val entity: Array<Entity>)
data class Entity(val id: Int, val name: String)
```
And I try to parse it: 
```Kotlin
@Test
fun arrayParse() {

    val array = """{
        "entity":[
            {"id": 1, "name": "foo"},
            {"id": 2, "name": "bar"}
        ]
     }"""

    Klaxon().parse<FooBar>(array)
}
```

I'll get the following error:
```
kotlin.reflect.jvm.internal.KotlinReflectionInternalError: Call is not yet supported for this function: public constructor Array<T>(size: kotlin.Int, init: (kotlin.Int) -> T) defined in kotlin.Array[DeserializedClassConstructorDescriptor@24269709] (member = null)
```

I'm new to Kotlin and I found out, that I just need to change `Array<Entity>` in my data class to `List<Entity>`. So maybe this is not an error at all and I'm just leaving this here for future generations, which may have the same problem I had.

**Comment by cbeust** [comment]
Yes, I'm aware.

The short version is: arrays are tricky.

The long version: Kotlin's `Array` maps to Java's arrays, which are reified and a complete corner case for reflection purposes. I'll try to see if `kotlin-reflect` helps there but I doubt it since it can't magically materialize information that's not present in Java's reflection in the first place.

In the meantime, use `ArrayList` (which you should anyway :-)).

**Comment by SteffenGorenflo** [comment]
Thanks for the response. And yes I will use `ArrayList` or `List`, I don't care :)

Keep up the good work 👍

**Comment by cbeust** [comment]
Fixed in 2.1.6.
